src/transactions/TransactionFrame.cpp

```diff
@@ -407,12 +588,19 @@ TransactionFrame::commonValidPreSeqNum(AbstractLedgerTxn& ltx, bool chargeFee,
         return false;
     }
 
-    if (!loadSourceAccount(ltx, header))
+    auto sourceAccount = loadSourceAccount(ltx, header);
+    if (!sourceAccount)
     {
         getResult().result.code(txNO_ACCOUNT);
         return false;
     }
 
+    if (isTooEarlyForAccount(header, sourceAccount, lowerBoundCloseTimeOffset))
```

Comment:
This is problematic for a couple reasons:
* this is too early: we need to consume the sequence number in that case (and avoid allowing it to be replayed later)
* This is also a CAP bug - we should use a different error code than `txTOO_EARLY` as `txTOO_EARLY` tells the clients that the transaction may be valid later and should retry later - but if we consumed the seqnum, this will never be true

Reply:
this situation can happen in what the CAP describes with
> Unfortunately, this does not entirely eliminate the possibility of non-executable transactions in blocks; for instance, a BUMP_SEQUENCE operation in a transaction from a different sourceAccount can invalidate the minSeqAge or minSeqLedgerGap on another transaction.

Reply:
Why id a tx not valid later when seqAge to now is less than minSeqAge?

Reply:
Oh I see. So when this condition fails during validation it is always too early, but when it fails during apply too early would be misleading because the tx is burned at that point. Is that correct?

Reply:
>this is too early: we need to consume the sequence number in that case (and avoid allowing it to be replayed later)

Also, could you elaborate on this? Why do we need to consume the seq num of the tx if we check if it is too early? That would break payment channels, and most uses of relative time locks.

Reply:
@leighmcculloch this is at apply time. Because we require relative time locks to be first in a block, we normally would not encounter this condition. This is only possible if somebody does a `BumpSequenceOp` in the same ledger and that transaction executes first.